src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx

```diff
@@ -0,0 +1,216 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Scroll from 'react-scroll';
+import appendQuery from 'append-query';
+import URLSearchParams from 'url-search-params';
+
+import AlertBox from '@department-of-veterans-affairs/jean-pants/AlertBox';
+import LoadingIndicator from '@department-of-veterans-affairs/jean-pants/LoadingIndicator';
+
+import { getScrollOptions } from '../../../platform/utilities/ui';
+
+import {
+  acceptTerms,
+  fetchLatestTerms,
+  fetchTermsAcceptance
+} from '../actions';
+
+const ScrollElement = Scroll.Element;
+const scroller = Scroll.scroller;
+
+const TERMS_NAME = 'mhvac';
+
+const unagreedBannerProps = {
+  headline: 'Using Vets.gov Health Tools',
+  content: 'Before you can use the health tools on Vets.gov, you’ll need to read and agree to the Terms and Conditions below. This will give us your permission to show you your VA medical information on this site.'
+};
+
+export class MhvTermsAndConditions extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const searchParams = new URLSearchParams(window.location.search);
+    this.redirectUrl = searchParams.get('tc_redirect');
+
+    this.state = {
+      isAgreementChecked: false,
+      showAcceptedMessage: false,
+      showCanceledMessage: false
+    };
+  }
+
+  componentDidMount() {
+    this.props.fetchLatestTerms(TERMS_NAME);
+    if (sessionStorage.userToken) { this.props.fetchTermsAcceptance(TERMS_NAME); }
+  }
+
+  redirect = () =>  {
+    if (this.redirectUrl) {
+      const newUrl = appendQuery(this.redirectUrl, { tc_accepted: true }); // eslint-disable-line camelcase
+      window.location.replace(newUrl);
+    }
+  }
+
+  handleAgreementCheck = () => {
+    this.setState({ isAgreementChecked: !this.state.isAgreementChecked });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.acceptTerms(TERMS_NAME);
+    this.setState({ showAcceptedMessage: true }, () => {
+      scroller.scrollTo('banner', getScrollOptions());
+      this.redirect();
+    });
+  }
+
```

Comment:
Would it be worth returning a promise from the `acceptTerms` action and using `setState` in a `.then`?  I ran it locally and the API endpoint wasn't configured so it returned a `404`, but it was kinda confusing how it still showed the success message.

Reply:
Good point. We probably don't want to show the success before the API returns with a success. Usually I would handle this in `componentDidUpdate`.

```
  componentDidUpdate() {
    if (!prevProps.accepted && this.props.accepted) {
      this.setState({ showAcceptedMessage: true }, () => {
        scroller.scrollTo('banner', getScrollOptions());
        this.redirect();
      });
    }
  }
```

Reply:
Ah nice! That's a better way to do it than my idea but I still miss the days where you just used Promises for situations like that.